shapes/profile-shape.ttl

```diff
@@ -0,0 +1,107 @@
+PREFIX sh: <http://www.w3.org/ns/shacl#>
+PREFIX dct: <http://purl.org/dc/terms/>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX solid: <http://www.w3.org/ns/solid/terms#>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+PREFIX org: <http://www.w3.org/ns/org#>
+PREFIX vcard: <http://www.w3.org/2006/vcard/ns#>
+PREFIX schema: <http://schema.org/>
+PREFIX foaf: <http://xmlns.com/foaf/0.1/>
+PREFIX : <#>
+
+:ProfileShape     a sh:NodeShape;
+    dct:title "Profile shape";
+    :property 
+            [
+        :datatype xsd:string;
+        :name "his/hers/theirs...";
+        :path solid:preferredRelativePronoun ],
+            [
+        :datatype xsd:string;
+        :name "him/her/them...";
+        :path solid:preferredObjectPronoun ],
+            [
+        :datatype xsd:string;
+        :name "he/she/they...";
+        :path solid:preferredSubjectPronoun ],
+            [
+        :name "Involvement with Organization";
+        :node  [
+            a sh:NodeShape;
+            :property  [
+                :datatype xsd:string;
+                :name "Describe your role";
+                :path schema:description ],
+                    [
+                :name "Organization"@en,
+                       "Organization"@fr;
+                :path org:organization ],
+                    [
+                :name "occupation";
+                :path org:role ],
+                    [
+                :datatype xsd:string;
+                :path vcard:role ] ];
+        :path [:inversePath org:member ] ],
+            [
+        :name "Skills";
+        :node  [
+            a sh:NodeShape;
+            :property  [
+                :name "skill";
+                :path solid:publicId ] ];
+        :path schema:skills ],
+            [
+        :name "Languages";
+        :path schema:knowsLanguage;
+        :property  [
+            :node  [
+                a sh:NodeShape;
+                :property  [
+                    :node  [
+                        a sh:NodeShape;
+                        :property  [
+                            :name "Language";
+                            :path solid:publicId ] ];
+                    :path  (
+                    [
+                            :zeroOrMorePath rdf:rest ]
+                   rdf:first ) ] ];
+            :path schema:knowsLanguage;
+            :targetClass rdf:Collection ] ],
```

Comment:
Not sure if this is correct per SHACL but as I understand schemaorg, expected range is Text or Language.

```suggestion
            :targetClass schema:Text, schema:Language ] ],
```

Reply:
Agreed (except that `:targetClass` should be `sh:class`). `schema:knowsLanguage` can just be in the profile multiple times, just as `schema:skills`, so no need for a complex `rdf:Collection` structure.

Reply:
The code uses the structure to allow you to put your languages in order.  I think that is quite important in practice.  People's ability in different language tends to vary a lot.

Reply:
Suggestions above seem good and useful.

---

I'll just leave a reminder about the following. Mentioned the Cognitive Characteristics Ontology ( https://smiy.sourceforge.net/cco/spec/cognitivecharacteristics.html ) elsewhere but it may serve as useful input to towards providing a solution in particular for example ( https://github.com/solid/webid-profile/issues/52 ):

>Factors specific to physical, physiological, genetic, mental, economic, cultural, social, or behavioural identity.

In my https://csarven.ca/cv , I say I have `cco:skill` and refer to the human-languages in ESCO. I didn't bother to order or put weights (and maybe that's overkill.. not sure.). The details seems useful for CV/Resume-like cases but not necessarily generic Profile info? Could go either way.. especially if the application can simply adapt the UI to use the user's preferred or most skilled language first, then try others (or even give option to change.) So, ordering languages can be simple to input and consume.

Reply:
Re Collections: the semantics of `schema:knowsLanguage` do not allow for that; the range consists of `schema:Language` and `schema:text`. If we want order/weight, I suggest we indeed go with something like CCO as @csarven mentioned, even though this might make profiles quite complex for the human reader.